atlasdb-cassandra-integration-tests/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueServiceIntegrationTest.java

```diff
@@ -125,6 +136,40 @@ public void testCreateTableCaseInsensitive() throws TException {
     }
 
     @Test
+    public void testGcGraceSecondsUpgradeIsApplied() throws TException {
+        CassandraKeyValueService kvs = createKvs(getConfigWithGcGraceSeconds(FOUR_DAYS_IN_SECONDS));
+        //first startup same as initial - no upgrade
+        verify(logger, times(1))
+                .debug(startsWith("No tables are being upgraded on startup. No updated table-related settings found."));
+        kvs.createTable(testTable, AtlasDbConstants.GENERIC_TABLE_METADATA);
+        assertThatGcGraceSecondsIs(kvs, FOUR_DAYS_IN_SECONDS);
+        kvs.close();
+
+        CassandraKeyValueService kvs2 = createKvs(getConfigWithGcGraceSeconds(ONE_HOUR_IN_SECONDS));
+        assertThatGcGraceSecondsIs(kvs2, ONE_HOUR_IN_SECONDS);
+        kvs.close();
+        //startup with different GC grace seconds - should upgrade
+        verify(logger, times(1))
+                .debug(startsWith("New table-related settings were applied on startup!!"));
+
+        CassandraKeyValueService kvs3 = createKvs(getConfigWithGcGraceSeconds(ONE_HOUR_IN_SECONDS));
+        assertThatGcGraceSecondsIs(kvs3, ONE_HOUR_IN_SECONDS);
+        //startup with same gc grace seconds as previous one - no upgrade
+        verify(logger, times(2))
+                .debug(startsWith("No tables are being upgraded on startup. No updated table-related settings found."));
+        kvs3.close();
+    }
+
+    private void assertThatGcGraceSecondsIs(CassandraKeyValueService kvs, int gcGraceSeconds) throws TException {
+        List<CfDef> knownCfs = kvs.getClientPool().runWithRetry(client ->
+                client.describe_keyspace("atlasdb").getCf_defs());
+        CfDef clusterSideCf = Iterables.getOnlyElement(knownCfs.stream()
+                .filter(cf -> cf.getName().equals(getInternalTestTableName()))
+                .collect(Collectors.toList()));
+        assertThat(clusterSideCf.gc_grace_seconds == gcGraceSeconds, is(true));
```

Comment:
nit: preferably `assertThat(clusterSideCf.gc_grace_seconds, isEqualTo(gcGraceSeconds))`; this one won't explain what happened if things go wrong